### PR TITLE
Add ability to comment

### PR DESCRIPTION
**Pull request by amuamushu**
This feature required adding an HTML form and a doPost method. The doPost method retrieves the text submitted by the HTML form and adds it the existing ArrayList containing the messages. 

An event listener is added to the HTML so that every time the page reloads, the JavaScript method getMessageFromJSON(), which fetches from /data, is called. 

Since doPost redirects to the current page after adding the message, all elements in the ArrayList appears in the DOM.

Styling for the comment box and additional inputs are intended to be added after step 6 is complete.

Comment Box Before Any Comments Were Added: https://screenshot.googleplex.com/uD3YirCTmT1.png
Comment Box After 1 Comment Was Added: https://screenshot.googleplex.com/aW1E5vd6o1v.png
Comment Box After Multiple Comments Were Added: https://screenshot.googleplex.com/HPwaENW4c9A.png

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -53,4 +49,14 @@ private String convertToJsonUsingGson(ArrayList toConvert) {
     String json = gson.toJson(toConvert);
     return json;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment-input");
```
Comment (billie-cat) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: Can you create a string for this and add a comment to what it means?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -53,4 +49,14 @@ private String convertToJsonUsingGson(ArrayList toConvert) {
...
+    messages.add(comment);
+
+    // Redirects to the current page to see new comment added.
+    response.sendRedirect("/index.html#connect-container");
```
Comment (billie-cat) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: Can you create a private static final string for this as well and add a comment.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (billie-cat) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: I meant a private static final string.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -169,6 +169,7 @@ function addMultipleMessagesToDom(messages) {
 
 /**
  * Fetches the message from the JSON server /data and adds it to the DOM.
+ *     Method is called everytime the page is refreshed.
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:172]: For this comment I would add a new line in-between as it is extra detail and can be in the next paragraph.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -84,7 +84,7 @@ function createPTag(text) {
 
 /**
  * Reveals hidden description upon mouseover and hides the project title
- *  text.
+ *     text.
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:87]: I am confused as to why there is extra spacing. Is this something asked for in the style guide?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (amuamushu) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done!

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (f-camp) [reply portfolio/src/main/webapp/script.js:172]: In that case if it is a new paragraph it should be prefixed with <p>

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: I think I got carried away with the spacing. Thank you for catching that! I think there should only be spacing for the tags. https://google.github.io/styleguide/jsguide.html#jsdoc-line-wrapping

Done!

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: Understood! Done!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (amuamushu) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I created a variable for the string. For adding a comment, I added a comment for the constants. Is that what you mean?

Comment (billie-cat) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Yes! Thank you

**Pull request merged**